app/(auth)/logout/logoutAction.ts

```diff
@@ -0,0 +1,18 @@
+'use server';
+import { cookies } from 'next/headers';
+import { deleteSessionByToken } from '../../../database/sessions';
+
+
+export default async function setCookie() {
+  const cookieStore = cookies();
+  const token = cookieStore.get('sessionToken');
+
+  if (token) {
+    await deleteSessionByToken(token.value);
+  }
+
+  (cookieStore as any).set('sessionToken', '', {
```

Comment:
probably `as unknown as ResponseCookies` is the correct type

https://github.com/vercel/next.js/blob/062fbf6d2d1629ffa240b87af35955c51f5eaeaa/packages/next/src/server/async-storage/request-async-storage-wrapper.ts#LL42C4-L42C19

until Shu fixes it:

- https://github.com/vercel/next.js/issues/49259#issuecomment-1540019323